File: elixir/6ku/Highest Scoring Word.ex
# Highest Scoring Word
# https://www.codewars.com/kata/57eb8fcdf670e99d9b000272

defmodule Kata do
  def high(str) do
    {_, word} =
      str
      |> String.split(" ")
      |> Enum.group_by(fn x -> String.to_charlist(x) |> Enum.reduce(0, &(&1 + &2 - 96)) end)
      |> Enum.max()

    List.first(word)
  end
end

File: ruby/6ku/Highest Scoring Word.rb
# frozen_string_literal: true

# Highest Scoring Word
# https://www.codewars.com/kata/57eb8fcdf670e99d9b000272/train/ruby
#
def high(x)
  x.split.max_by { |word| word.sum - 96 * word.length }
end

# p high("man i need a taxi up to ubud") #== "taxi"
p high("what time are we climbing up the volcano") #== "taxi"
# p high("what time are we climbing up the volcano") == "volcano"
# p high("take me to semynak") == "semynak"
# p high("aa b") == "b"
# p high("b aa") == "b"
p high("b aa") #== "b"
